assets/src/components/activities/likert/reports/LikerReportRenderer.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { VegaLite, VisualizationSpec } from 'react-vega';
+import { makeRequest } from 'data/persistence/common';
+
+export interface LikerReportRendererProps {
+  sectionId: string;
+  activityId: string;
+  sectionSlug: string;
+}
+
+export type Parent = {
+  title: string;
+  slug?: string;
+};
+
+export type Report = {
+  type: 'success';
+  spec?: VisualizationSpec;
+  prompts?: string;
+  message?: string;
+  parent: Parent;
+};
+
+export const LikerReportRenderer = (props: LikerReportRendererProps) => {
```

Comment:
I just noticed this is misspelled `Liker` instead of `Likert`

Reply:
I shifted all DB work from the `Oli.Rendering.Report.Html` report function. That way, rendering of any page containing the report is fast. The actual contents of the activity are rendered when the client-side LikertReportRenderer calls back to the `activity_report_data_controller` to fetch data in addition to other content that requires DB calls. I figured that I may as well bundle the data for the VegaLite with the spec for this activity report. Other activities may choose to use the `activity_report_data_controller` otherwise.